.github/workflows/test.yml

```diff
@@ -0,0 +1,82 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    - cron: '0 6 * * *'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-2019
+          - ubuntu-latest
+          # got "installer: Error - The FUSE for macOS installation package is not compatible with this version of macOS."
+          # from osxfuse
+          # - macos-latest
+          - macos-11
+        python:
+          - 3.7
+          - 3.8
+          - 3.9
+          - '3.10'  # Needs quotes so YAML doesn't think it's 3.1
+          - '3.11'
+
+    steps:
+    - name: Set up environment
+      uses: actions/checkout@v3
+
+    - name: Set up Python ${{ matrix.python }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python }}
+
+    - name: Install Python dependencies and refuse itself
+      run: |
+        python -m pip install --upgrade pip wheel
+        pip install ".[dev]"
+
+    - name: Install FUSE (Linux)
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt-get update -qq
+        sudo eatmydata apt-get install -y fuse
+
+    - name: Install OSXFUSE (MacOS)
+      if: startsWith(matrix.os, 'macos')
+      run: |
+        brew update
+        brew install osxfuse
```

Comment:
attempt to run tests with this seems to just stall on macos-11 without any feedback: see https://github.com/yarikoptic/refuse/actions/runs/4559603821/jobs/8043727829 (still running ,will check if outputs anything tomorrow)